### PR TITLE
parse att&ck for output doc

### PR DESCRIPTION
**Pull request by mr-tz**
Add ATT&CK like MBC (#576)

#526

### Checklist

- [ ] No CHANGELOG update needed

- [ ] No new tests needed

- [ ] No documentation update needed

### CODE REVIEW

**Comment by williballenthin** [comment]
image

when a field's data is not present, should the field:

  1. contain an empty string (like above), or
  2. not be present

client code would look like:


## 1
```py
if match["att&ck"]["subtechnique"] != "":
    print(...)
```

## 2
```py
if "subtechnique" in match["att&ck"]:
    print(...)
```

File: tests/test_render.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    assert attack["id"] == id
+    assert attack["tactic"] == tactic
+    assert attack["technique"] == technique
+    assert attack["subtechnique"] == subtechnique
+
+    assert format_parts_id(attack) == canonical
+
+
+def test_render_meta_mbc():
```
Comment (williballenthin) [inline tests/test_render.py:41]: thanks for adding these

File: tests/test_rules.py [hunk 2]
```diff
@@ -399,6 +399,34 @@ def test_invalid_rules():
             )
         )
 
+    # att&ck and mbc must be lists
```
Comment (williballenthin) [inline tests/test_rules.py:402]: 👍🏼

File: capa/render/__init__.py [hunk 3]
```diff
@@ -163,11 +163,44 @@ def convert_match_to_result_document(rules, capabilities, result):
...
 
 def convert_meta_to_result_document(meta):
+    attacks = meta.get("att&ck", [])
+    meta["att&ck"] = [parse_canonical_attack(attack) for attack in attacks]
     mbcs = meta.get("mbc", [])
     meta["mbc"] = [parse_canonical_mbc(mbc) for mbc in mbcs]
     return meta
 
 
+def parse_canonical_attack(attck):
```
Comment (williballenthin) [inline capa/render/__init__.py:173]: some code dup here but i think thats just fine. these code is quite readable and unlikely to change often.

**Review by williballenthin — APPROVED** [review]

File: capa/render/__init__.py (hunk 3 above)
Comment (mr-tz) [reply capa/render/__init__.py]: improved in 63cd700

**Comment by mr-tz** [comment]
leaving the json as is (1.)

**Pull request merged**